src/components/BarDiagram/index.js

```diff
@@ -72,36 +77,33 @@ class BarDiagram extends Component {
             </Animate>
           </Layer>
         </Chart>
-        <center><p style={{fontSize:"1.0em"}}>
-          <span style={{color:"#33ff66"}}> WordCount </span>
-        </p></center>
+        <div style={{position: "center", fontSize: "1.2em", textAlign: "center"}}>
+          {generateLegend(rawAll)}
+        </div>
       </div>
     );
   }
 }
 
+//Generate the legend based on the current data
+function generateLegend(data) {
+  var result = [];
+  for(var i=0; i<data.length; i++) {
```

Comment:
Easier to use data.map instead of a for loop. Just a style suggestion, doesnt affect functionality.